auth/auth.js

```diff
@@ -41,7 +43,9 @@ export function authenticateToken(req) {
     });
 }
 
-function extractTokenFromHeader(req) {
-    const authHeader = req.headers["authorization"];
+export function extractTokenFromHeader(req) {
+    const authHeader = req.hasOwnProperty("headers")
+        ? req.headers["authorization"]
+        : undefined;
```

Comment:
@miczoc98 notice changes in your original implementation. If req didn't have headers property, it was throwing an exception, which you probably didn't even know about.
UTs allowed us to detect it, but it's for you to decide how we should cope with this. If you think that it's unreasonable to expect your function to work with this kind of input and you can assure that such input is improbable, you don't have to change anything (but adjust UTs so they won't fail).
Accepting these changes would make function safer and easier to use elsewhere - but again, it's up to you, because you could argue that "You aren't gonna need it" (YAGNI).